packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1521,6 +1566,61 @@ export class TransactionController extends BaseController<
   ): boolean {
     return remoteGasUsed !== localGasUsed;
   }
+
+  private requestApproval(txMeta: TransactionMeta) {
+    const id = this.getApprovalId(txMeta);
+    const { origin } = txMeta;
+    const type = ApprovalType.Transaction;
+
+    this.messagingSystem
+      .call(
+        'ApprovalController:addRequest',
+        {
+          id,
+          origin: origin || ORIGIN_METAMASK,
+          type,
+          requestData: {
+            id: txMeta.id,
+            origin: origin || ORIGIN_METAMASK,
+            transaction: {
+              ...txMeta.transaction,
```

Comment:
It is safer to include the entire `txMeta`?

Reply:
Removed TxMeta only including `{ txId: txMeta.id }` as in the extension.